pkg/api/v1/stub.go

```diff
@@ -50,9 +60,37 @@ func (g *StubGroup) ListStubs(ctx echo.Context) error {
 		return HTTPBadRequest("Failed to decode query parameters")
 	}
 
-	if stubs, err := g.backendRepo.ListStubs(ctx.Request().Context(), filters); err != nil {
-		return HTTPInternalServerError("Failed to list stubs")
+	if filters.Pagination {
+		if stubs, err := g.backendRepo.ListStubsPaginated(ctx.Request().Context(), filters); err != nil {
+			return HTTPInternalServerError("Failed to list stubs")
+		} else {
+			return ctx.JSON(http.StatusOK, stubs)
+		}
 	} else {
-		return ctx.JSON(http.StatusOK, stubs)
+		if stubs, err := g.backendRepo.ListStubs(ctx.Request().Context(), filters); err != nil {
+			return HTTPInternalServerError("Failed to list stubs")
+		} else {
+			return ctx.JSON(http.StatusOK, stubs)
+		}
 	}
 }
+
+func (g *StubGroup) RetrieveStub(ctx echo.Context) error {
+	stubID := ctx.Param("stubId")
+	c, _ := ctx.(*auth.HttpAuthContext)
+
+	stub, err := g.backendRepo.GetStubByExternalId(ctx.Request().Context(), stubID)
+	if err != nil {
+		log.Println(err)
+		return HTTPInternalServerError("Failed to retrieve stub")
+	} else if stub == nil {
+		return HTTPNotFound()
+	}
+
+	if stub.Workspace.ExternalId != c.AuthInfo.Workspace.ExternalId && c.AuthInfo.Token.TokenType != types.TokenTypeClusterAdmin {
```

Comment:
Can we check for this before the call to GetStubByExternalId? We have the stubid from the request and the token type.

Reply:
NVM, I misread.